### PR TITLE
Handle decimal conversion and assert emitted event

### PR DESCRIPTION
**Pull request by longbowlu**

### CODE REVIEW

File: contracts/SuiBridge.sol [hunk 1]
```diff
@@ -209,33 +222,88 @@ contract SuiBridge is
...
+        if (suiDecimal == ethDecimal) {
+            return uint256(originalAmount);
+        }
+
+        // Safe guard for the future
+        require(ethDecimal > suiDecimal, "Eth decimal should be larger than sui decimal");
+
+        // TODO: do we worry about overflow here?
+        uint256 factor = 10**(ethDecimal - suiDecimal); // Difference in decimal places
+        uint256 newAmount = originalAmount * factor;
```
Comment (longbowlu) [inline contracts/SuiBridge.sol:285]: is this safe?

File: contracts/SuiBridge.sol [hunk 2]
```diff
@@ -209,33 +222,88 @@ contract SuiBridge is
...
             destinationChainId,
             Messages.ETH,
-            amount,
+            suiAdjustedAmount,
             msg.sender,
             targetAddress
         );
+
+        // increment token transfer nonce
+        nonces[Messages.TOKEN_TRANSFER]++;
```
Comment (longbowlu) [inline contracts/SuiBridge.sol:246]: move it here so the first nonce will be 0 (otherwise it's 1)

File: contracts/SuiBridge.sol [hunk 3]
```diff
@@ -209,33 +222,88 @@ contract SuiBridge is
...
+            require(originalAmount <= type(uint64).max, "Amount too large for uint64");
+            return uint64(originalAmount);
+        }
+
+        // Safe guard for the future
+        require(ethDecimal > suiDecimal, "Eth decimal should be larger than sui decimal");
+
+        uint256 factor = 10**(ethDecimal - suiDecimal); // Difference in decimal places
+        // TODO: do we worry about underflow here?
+        uint256 newAmount = originalAmount / factor;
```
Comment (longbowlu) [inline contracts/SuiBridge.sol:264]: safe?

File: contracts/SuiBridge.sol [hunk 4]
```diff
@@ -189,18 +200,20 @@ contract SuiBridge is
...
             destinationChainId,
             tokenId,
-            amount,
+            suiAdjustedAmount,
             msg.sender,
             targetAddress
         );
+
+        // increment token transfer nonce
+        nonces[Messages.TOKEN_TRANSFER]++;
```
Comment (longbowlu) [inline contracts/SuiBridge.sol:216]: move it here so the first nonce will be 0 (otherwise it's 1)

File: test/BridgeVaultTest.t.sol [hunk 5]
```diff
@@ -91,6 +91,6 @@ contract BridgeVaultTest is Test, BridgeVault {
 
         // Check that the burn was successful
         assertEq(token.balanceOf(bob), 400);
-        assertEq(token.balanceOf(address(vault)), 600);
+        assertEq(token.balanceOf(address(vault)), 500);
```
Comment (longbowlu) [inline test/BridgeVaultTest.t.sol:94]: The test fails on main. I think this should be 500 ?

File: contracts/SuiBridge.sol [hunk 6]
```diff
@@ -166,6 +176,7 @@ contract SuiBridge is
         nonces[Messages.BRIDGE_UPGRADE]++;
     }
 
+    // TODO: add checks for destination chain ID. Disallow invalid values
```
Comment (longbowlu) [inline contracts/SuiBridge.sol:179]: For example one can't send sepolia eth to sui mainnet.

Comment (Bridgerz) [reply contracts/SuiBridge.sol:179]: good call.

File: contracts/SuiBridge.sol (hunk 4 above)
Comment (Bridgerz) [reply contracts/SuiBridge.sol:216]: nice catch

File: contracts/SuiBridge.sol (hunk 3 above)
Comment (Bridgerz) [reply contracts/SuiBridge.sol:264]: may want to use the "SafeMath" library from OZ. But it looks like 260 should catch underflow.

File: contracts/SuiBridge.sol (hunk 1 above)
Comment (Bridgerz) [reply contracts/SuiBridge.sol:285]: same response as previous.

File: test/BridgeVaultTest.t.sol (hunk 5 above)
Comment (Bridgerz) [reply test/BridgeVaultTest.t.sol:94]: correct and also this test is useless....

**Pull request merged**